// @vincjo/datatables/core/Comparator.js
const stringify = (value = null) => {
    return String(value)
        .toLowerCase()
        .normalize("NFD")
        .replace(/[\u0300-\u036f]/g, "");
};
const isNull = (entry) => {
    if (entry === null || entry === undefined)
        return true;
};
export const check = {
    contains: (entry, value) => {
        return stringify(entry).indexOf(stringify(value)) > -1;
    },
    startsWith: (entry, value) => {
        return stringify(entry).startsWith(stringify(value));
    },
    endsWith: (entry, value) => {
        return stringify(entry).endsWith(stringify(value));
    },
    isEqualTo: (entry, value) => {
        return stringify(entry) === stringify(value);
    },
    isGreaterThan: (entry, value) => {
        if (isNull(entry))
            return false;
        return entry > value;
    },
    isGreaterThanOrEqualTo: (entry, value) => {
        if (isNull(entry))
            return false;
        return entry >= value;
    },
    isLessThan: (entry, value) => {
        if (isNull(entry))
            return false;
        return entry < value;
    },
    isLessThanOrEqualTo: (entry, value) => {
        if (isNull(entry))
            return false;
        return entry <= value;
    },
    isBetween: (entry, value) => {
        if (isNull(entry))
            return false;
        const [min, max] = value;
        return entry >= min && entry <= max;
    },
    isStrictlyBetween: (entry, value) => {
        if (isNull(entry))
            return false;
        const [min, max] = value;
        return entry > min && entry < max;
    },
    isTrue: (entry, value = 'boolean') => {
        return entry === true;
    },
    isFalse: (entry, value = 'boolean') => {
        return entry === false;
    },
    isNull: (entry, value = 'null') => {
        return entry === null || entry === undefined;
    },
    isNotNull: (entry, value = 'null') => {
        return (entry === null || entry === undefined) ? false : true;
    },
};
